src/Worker/WorkerResolver.php

```diff
@@ -0,0 +1,37 @@
+<?php
+declare(strict_types=1);
+
+namespace Baldinof\RoadRunnerBundle\Worker;
+
+use Spiral\RoadRunner\Environment\Mode;
+use Spiral\RoadRunner\EnvironmentInterface;
+
+final class WorkerResolver implements WorkerResolverInterface
+{
+    private EnvironmentInterface $environment;
+    private HttpWorker $httpWorker;
+    private TemporalWorker $temporalWorker;
+
+    public function __construct(
+        EnvironmentInterface $environment,
+        HttpWorker $httpWorker,
+        TemporalWorker $temporalWorker
+    )
+    {
+        $this->environment = $environment;
+        $this->httpWorker = $httpWorker;
+        $this->temporalWorker = $temporalWorker;
+    }
+
+    public function resolve(string $mode): WorkerInterface
+    {
+        if ($this->environment->getMode() === Mode::MODE_HTTP) {
+            return $this->httpWorker;
+        }
+        if ($this->environment->getMode() === Mode::MODE_TEMPORAL) {
+            return $this->temporalWorker;
+        }
+
+        throw new \RuntimeException('Could not resolve worker.');
```

Comment:
Maybe a dedicated exception here, and the message could display expected and got values for `$this->environment->getMode()`